### PR TITLE
Feature/uweb 6142 article labels

### PR DESCRIPTION
**Pull request by dan-entous**
* [x] add labels to `card-article`

image

### CODE REVIEW

File: src/js/components/molecules/cards/card-article/card-article.scss [hunk 1]
```diff
@@ -29,9 +36,23 @@
     margin-bottom: $buildingUnit * 2;
 }
 
-.card-article__description {
-    font-size: 14px;
-    margin-bottom: $buildingUnit;
-    padding-left: $buildingUnit;
-    padding-right: $buildingUnit;
+.card-article__list {
```
Comment (matijs) [inline src/js/components/molecules/cards/card-article/card-article.scss]: `list` is too generic I think… `labels-list` maybe? Bit longer but more descriptive.

File: src/js/components/molecules/cards/card-article/card-article.scss [hunk 2]
```diff
@@ -29,9 +36,23 @@
...
+
+    // offset margin-bottom on the list items in the last row of the flexbox
+    margin-top: -$buildingUnit / 2;
+
+    // offset margin-right on the list items in the last column of the flexbox
+    margin-left: -$buildingUnit / 2;
+
+    .list__item {
+        line-height: 15px;
+        margin-right: $buildingUnit / 2;
```
Comment (matijs) [inline src/js/components/molecules/cards/card-article/card-article.scss]: probably `white-space: nowrap`?

File: src/js/components/molecules/cards/card-article/card-article.scss [hunk 3]
```diff
@@ -29,9 +36,23 @@
...
+    margin-bottom: $buildingUnit / 2;
+
+    // offset margin-bottom on the list items in the last row of the flexbox
+    margin-top: -$buildingUnit / 2;
+
+    // offset margin-right on the list items in the last column of the flexbox
+    margin-left: -$buildingUnit / 2;
+
+    .list__item {
+        line-height: 15px;
```
Comment (matijs) [inline src/js/components/molecules/cards/card-article/card-article.scss]: :( px on a `line-height`?

File: src/js/components/molecules/cards/card-article/card-article.ts [hunk 4]
```diff
@@ -9,8 +9,29 @@ import WithRender from './card-article.vue.html';
...
-    @Prop({ type: String }) description?: CardArticleProps['description'];
-    @Prop({ type: String }) url?: CardArticleProps['url'];
+    @Prop({ type: String }) description!: CardArticleProps['description'];
+    @Prop({ type: Array }) labels?: CardArticleProps['labels'];
+    @Prop({ type: String }) url!: CardArticleProps['url'];
+
+    get listItems(): object {
+        let properties: any = [];
+
+        if (this.labels) {
```
Comment (matijs) [inline src/js/components/molecules/cards/card-article/card-article.ts]: with a default empty array, this check becomes unnecessary

File: src/js/components/molecules/cards/card-article/card-article.vue.html [hunk 5]
```diff
@@ -9,4 +9,5 @@ <h3 class="card-article__heading">{{ title }}</h3>
     <p v-if="description" class="card-article__description">
         {{ description }}
     </p>
+    <list-unordered class="card-article__list" :listItems="listItems"/>
```
Comment (matijs) [inline src/js/components/molecules/cards/card-article/card-article.vue.html]: So yeah, mabye just a bunch of `<span>`s in a `<p>`?

File: src/js/components/molecules/cards/card-article/card-article.ts [hunk 6]
```diff
@@ -9,8 +9,29 @@ import WithRender from './card-article.vue.html';
 @Component({
     name: 'CardArticle',
 })
-
 export class CardArticle extends CardBase implements CardArticleProps {
-    @Prop({ type: String }) description?: CardArticleProps['description'];
-    @Prop({ type: String }) url?: CardArticleProps['url'];
+    @Prop({ type: String }) description!: CardArticleProps['description'];
+    @Prop({ type: Array }) labels?: CardArticleProps['labels'];
```
[the remark this answers is not in the document]
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.ts]: don't see a need for it ...

File: src/js/components/molecules/cards/card-article/card-article.scss [hunk 7]
```diff
@@ -29,9 +36,23 @@
...
-    margin-bottom: $buildingUnit;
-    padding-left: $buildingUnit;
-    padding-right: $buildingUnit;
+.card-article__list {
+    display: flex;
+    align-items: flex-start;
+    flex-wrap: wrap;
+    align-content: space-between;
+
+    padding-right: 4px;
```
[the remark this answers is not in the document]
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.scss]: ⚡️

File: src/js/components/molecules/cards/card-article/card-article.scss (hunk 1 above)
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.scss]: ⚡️

File: src/js/components/molecules/cards/card-article/card-article.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.ts]: can't figure out how to get typescript to agree.

File: src/js/components/molecules/cards/card-article/card-article.scss (hunk 2 above)
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.scss]: discussed with matijs.

File: src/js/components/molecules/cards/card-article/card-article.scss (hunk 3 above)
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.scss]: discussed with matijs.

File: src/js/components/molecules/cards/card-article/card-article.ts (hunk 4 above)
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.ts]: can't figure out how to get typescript to agree.

File: src/js/components/molecules/cards/card-article/card-article.vue.html (hunk 5 above)
Comment (dan-entous) [reply src/js/components/molecules/cards/card-article/card-article.vue.html]: can't figure out how to get typescript to agree.

**Pull request merged**